### ISSUE TITLE
Auto-Rescheduling of Pods based on availability of higher priority instance group nodes

### ISSUE THREAD

**Issue by tarun-asthana** [kubernetes/autoscaler#4697]
Hi All!
**Background:**
I am using priority expanding in cluster autoscaling for a kops cluster in AWS. Since AWS spot instance availability is an issue sometimes - we have another ondemand instance group with the same taints but lesser priority on autoscaler.
However, I want the pods to reschedule themselves automatically to spot instances once they become available again. I am running a lot of them and ondemand is pretty expensive. 

**Feature:**
Need auto rescheduling of pods to happen if a higher priority instance group node becomes available. Additionally lower priority nodes should get deleted once rescheduling is complete.

**Comment by pavel-z** [comment]
We have the same issue. 
We have spot instances as a primary class and fallback on-demand instances (with significantly lower priority). Sometimes on-demand instances are launched with autoscaler is unable to start spot instances. Such instance keeps running until it's terminated in the scope of the scale-down process, or some manual/external actions are applied.
It would be awesome if the controller had the possibility to replace nodes based on configured priority.
